src/Chunk.ts

```diff
@@ -280,14 +282,20 @@ export default class Chunk {
 		options: OutputOptions,
 		existingNames: Record<string, any>
 	): string {
-		const sanitizedId = sanitizeFileName(this.orderedModules[0].id);
+		const id = this.orderedModules[0].id;
+		const sanitizedId = sanitizeFileName(id);
 
 		let path: string;
-		if (isAbsolute(this.orderedModules[0].id)) {
+		if (isAbsolute(id)) {
+			const extension = extname(id);
+			console.log({ COMMON_JS_EXTENSIONS, extension });
 			const name = renderNamePattern(
-				options.entryFileNames || '[name].js',
+				options.entryFileNames ||
+					(COMMON_JS_EXTENSIONS.includes(extension) ? '[name].js' : '[name].[ext].js'),
 				'output.entryFileNames',
 				{
+					ext: () => extension.substr(1),
+					extname: () => extension,
```

Comment:
Would be nice to also have a test that directly uses the new pattern elements in `entryFileNames`. Could also be an extension of an existing test.

Reply:
Sure, gonna tweak it.